### PR TITLE
Provide ability to override kafka-monitor destination configs

### PR DESCRIPTION
**Pull request by dalassi1**
The main change here is treating the configs with prefix `destination.` as an override to the `consumer.` configs. This means that the connection to the _destination_ cluster's configs will be the combination of the default consumer configs with any override set using the prefix `destination.`.

### CODE REVIEW

File: src/main/java/com/salesforce/mirus/MirusSourceTask.java [hunk 1]
```diff
@@ -211,7 +209,8 @@ private void seekToOffsets(List<TopicPartition> partitionIds) {
   }
 
   private boolean isSkippedRecord(ConsumerRecord<byte[], byte[]> consumerRecord) {
-    TopicPartition topicPartition = new TopicPartition(consumerRecord.topic(), consumerRecord.partition());
+    TopicPartition topicPartition =
```
Comment (pdavidson100) [inline src/main/java/com/salesforce/mirus/MirusSourceTask.java:212]: Better to avoid mixing code changes with reformatting. Are these changes required by the "google-java-format" plugin in Default Google Java Style mode?  If so, maybe do formatting in a different PR?

File: src/main/java/com/salesforce/mirus/config/SourceConfig.java [hunk 2]
```diff
@@ -43,8 +43,17 @@ public String getTopicsRegex() {
...
-    return simpleConfig.getString(SourceConfigDefinition.DESTINATION_BOOTSTRAP_SERVERS.key);
+  private Map<String, Object> getDestinationProperties() {
+    return simpleConfig.originalsWithPrefix("destination.");
+  }
+
+  public Map<String, Object> getDestinationConsumerConfigs() {
+    Map<String, Object> reconciledConsumerProperties = getConsumerProperties();
+    Map<String, Object> destinationProperties = getDestinationProperties();
+    // override consumer properties with any KafkaMonitor-specific property
+    destinationProperties.forEach((k, v) -> reconciledConsumerProperties.put(k, v));
```
Comment (pdavidson100) [inline src/main/java/com/salesforce/mirus/config/SourceConfig.java:54]: Maybe just do `getDestinationProperties().forEach((k, v)`?

File: src/main/java/com/salesforce/mirus/config/TaskConfigDefinition.java [hunk 3]
```diff
@@ -8,13 +8,11 @@
 
 package com.salesforce.mirus.config;
 
+import com.salesforce.mirus.config.TaskConfig.ReplayPolicy;
```
Comment (pdavidson100) [inline src/main/java/com/salesforce/mirus/config/TaskConfigDefinition.java:11]: Unnecessary reformatting?

**Review by pdavidson100 — COMMENTED** [review]
Change look good. I have some minor comments. Also,I think should update the configuration section in the README.md. Also, is  it a concern that there are other non-consumer settings with the destination prefix? I think this will lead to warnings? E.g. see:
https://github.com/salesforce/mirus/blob/9eab3f69a4a9125b9bf9c9fd77949903f94a67f9/src/main/java/com/salesforce/mirus/config/SourceConfigDefinition.java#L48

File: src/main/java/com/salesforce/mirus/MirusSourceTask.java (hunk 1 above)
Comment (dalassi1) [reply src/main/java/com/salesforce/mirus/MirusSourceTask.java:212]: Yes this caused by the Google Java Style formatter. We can/should do that in a later PR.

File: src/main/java/com/salesforce/mirus/config/SourceConfig.java [hunk 4]
```diff
@@ -43,8 +44,23 @@ public String getTopicsRegex() {
     return simpleConfig.originalsWithPrefix("consumer.");
   }
 
-  public String getDestinationBootstrapServers() {
-    return simpleConfig.getString(SourceConfigDefinition.DESTINATION_BOOTSTRAP_SERVERS.key);
+  private Map<String, Object> getDestinationProperties() {
+    // handle destination.bootstrap.server separately
```
Comment (dalassi1) [inline src/main/java/com/salesforce/mirus/config/SourceConfig.java:48]: Kept this here so that these changes are backward compatible. See Line 53 which overrides only if its equivalent `destination.consumer.` prop is not defined.

**Comment by dalassi1** [comment]
@pdavidson100  I believe I addressed all the comments. Should be good for another review.

**Comment by pdavidson100** [comment]
Note that this change builds on https://github.com/salesforce/mirus/pull/52, and I think resolves the concern I raised in my review there.

File: src/main/java/com/salesforce/mirus/config/SourceConfig.java [hunk 5]
```diff
@@ -43,8 +44,23 @@ public String getTopicsRegex() {
...
+    String destBootstrap =
+        simpleConfig.getString(SourceConfigDefinition.DESTINATION_BOOTSTRAP_SERVERS.key);
+    Map<String, Object> destConsumerProps =
+        simpleConfig.originalsWithPrefix("destination.consumer.");
+    destConsumerProps.computeIfAbsent(
+        "bootstrap.servers", s -> destConsumerProps.put("bootstrap.servers", destBootstrap));
+    return destConsumerProps;
+  }
+
+  public Map<String, Object> getDestinationConsumerConfigs() {
```
Comment (pdavidson100) [inline src/main/java/com/salesforce/mirus/config/SourceConfig.java:58]: I think merging the source and destination consumer properties is really a "destination consumer" implementation detail, so I think this logic should live in `newDestinationConsumer()`. That way SourceConfig doesn't know anything about destination consumer.

File: src/main/java/com/salesforce/mirus/config/SourceConfig.java (hunk 4 above)
Comment (pdavidson100) [reply src/main/java/com/salesforce/mirus/config/SourceConfig.java:48]: I think we should add a comment making it clear we're doing this to maintain backward compatibility.

File: src/main/java/com/salesforce/mirus/config/SourceConfigDefinition.java [hunk 6]
```diff
@@ -91,11 +91,11 @@
       ConfigDef.Importance.MEDIUM,
       "Converter class to apply to source record headers"),
   DESTINATION_BOOTSTRAP_SERVERS(
-      "destination.bootstrap.servers",
-      ConfigDef.Type.STRING,
-      "",
-      ConfigDef.Importance.HIGH,
-      "Comma-separated list of destination bootstrap server endpoints in standard format. This is used by the Kafka Monitor for run-time topic validation");
+          "destination.bootstrap.servers",
```
Comment (pdavidson100) [inline src/main/java/com/salesforce/mirus/config/SourceConfigDefinition.java:94]: Should we add a "deprecated" warning here?

**Review by pdavidson100 — COMMENTED** [review]
Looking good, but a few more minor issues.

**Review by pdavidson100 — APPROVED** [review]
Looks good. A few small things: since "destination.bootstrap.servers" is deprecated we update `README.md` to reflect the change in the quickstart instructions. We also have a security alert on `junit` so we should update to `<version>[4.13.1,)</version>`.

**Review by pdavidson100 — APPROVED** [review]
LGTM
